module-bluetooth/Bluetooth/interface/profiles/HFP/HFP.cpp

```diff
@@ -422,18 +419,10 @@ namespace bluetooth
         return sco->getStreamData();
     }
 
-    void HFP::HFPImpl::start()
-    {
-        if (!isConnected) {
-            connect();
-        }
-        hfp_ag_set_speaker_gain(aclHandle, 8);
-        hfp_ag_set_microphone_gain(aclHandle, 10);
-    }
-
-    void HFP::HFPImpl::stop()
+    void HFP::HFPImpl::terminateCall() const noexcept
     {
         hfp_ag_terminate_call();
```

Comment:
Do we have a possibility to distinguish between situations when call was answered, then ended and just rejected? As far as I understand, these two cases should be handled differently - when call was answered and then we want to end it, it should be terminated by hfp_ag_terminate_call(). If call was rejected without prior answering, it should be terminated via hfp_ag_call_dropped().

Explicitly calling hfp_ag_release_audio_connection() (line below) should not be needed when the logic is handled as I suggested...at least I hope so :P

Reply:
we have more states that we should execute :
![image](https://user-images.githubusercontent.com/23299649/164445361-7db5f228-7925-4989-8d37-a6b6a53e2ebe.png)

Those should be connected properly with cellular events based on the state machine diagram that you created in the MOS-366 task. 
I don't want to add next x flags to handle call logic when it should be properly used with events. I just added the terminateCall method as a complementary to initiateCall, as the audio profile has start/stop methods.

Reply:
Ok, if it works, let's leave it as it is for now. I'll finish this state machine task ASAP and then we'll do it the way it is supposed to be done, not by adding more and more flags as a workaround.